### PR TITLE
Bump `google-protobuf` to v4.27

### PR DESCRIPTION
**Pull request by cocoahero**
The gem currently has a version lock on `google-protobuf` that prevents users from migrating to the `4.x` series of versions. This prevents us from upgrading our `protoc` version beyond v25, which includes needed support for service custom options, etc.

There shouldn't be any issues with supporting both `3.x` and `4.x` even though the code that `protoc` generates is not backwards compatible. All `twirp` currently cares about is that the types exist.

### CODE REVIEW

File: twirp.gemspec [hunk 1]
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |spec|
   spec.require_paths = ["lib"]
 
   spec.required_ruby_version = '>= 1.9'
-  spec.add_runtime_dependency 'google-protobuf', '~> 3.0', '>= 3.7.0'
```
Comment (arthurnn) [inline twirp.gemspec:22]: wouldnt this only allow 4.x versions? and not 3.x anymore?

Comment (cocoahero) [reply twirp.gemspec:22]: Fair point. I have updated the PR to allow for this.

File: twirp.gemspec [hunk 2]
```diff
@@ -19,7 +19,7 @@ Gem::Specification.new do |spec|
...
+  spec.add_runtime_dependency 'google-protobuf', '>= 3.7.0', '< 5'
```
Comment (colinbendell) [inline twirp.gemspec:22]: This should be `>= 3.18` (~2021 era) which is the first version that supports the descriptor syntax. Ideally we would use `>= 3.25` as [grpc does](https://github.com/grpc/grpc/pull/36982).

```suggestion
  spec.add_runtime_dependency 'google-protobuf', '>= 3.25', '< 5.a'
```

**Review by colinbendell — APPROVED** [review]

**Comment by ryanseys** [comment]
Would be awesome to get this landed so we can use `google-protobuf` 4.x 🎂 

Thanks @colinbendell for the PR!

**Pull request merged**